client/Packages/com.beamable/Runtime/Modules/Purchasing/UnityBeamablePurchaser.cs

```diff
@@ -58,24 +58,30 @@ public Promise<Unit> Initialize(IDependencyProvider provider)
 					return _initPromise;
 				}
 
+				
+				
 #if USE_STEAMWORKS && !UNITY_EDITOR
                 var builder = ConfigurationBuilder.Instance(new Steam.SteamPurchasingModule(_serviceProvider));
                 foreach (var sku in rsp.skus.definitions)
                 {
-                    builder.AddProduct(sku.name, ProductType.Consumable, new IDs
-                    {
-                        { sku.productIds.steam, Steam.SteamStore.Name }
-                    });
+				   if (sku == null) continue;
+
+                   var ids = UnityBeamablePurchaserUtil.CreateIdsFromSku(
+						new SkuIdPair {skuProductId = sku.productIds?.steam, storeId = Steam.SteamStore.Name,}

```

Comment:
What happens when `sku.productIds` is null?

Reply:
Having reasoned it out with the Play Store and iTunes cases, it just skips the ones that are null by virtue of not having a product ID for Steam. It is perfectly valid to have some SKUs for Steam and different SKUs for Apple and Google, so this pattern is fine (I guess; see my complaint about the iTunes and Android ones for details on my stance)